common/oauthTokenManager.go

```diff
@@ -300,38 +326,111 @@ func (uotm *UserOAuthTokenManager) UserLogin(tenantID, activeDirectoryEndpoint s
 	return nil
 }
 
-// getCachedTokenInfo get a fresh token from local disk cache.
-// If access token is expired, it will refresh the token.
-// If refresh token is expired, the method will fail and return failure reason.
-// Fresh token is persisted if access token or refresh token is changed.
+// getCachedTokenInfo retrieves a fresh token.
+// If a valid AuthenticationRecord exists, the token is obtained from the azidentity token cache.
+// If the access token has expired, it is refreshed.
+// If the refresh token has expired, the method fails and returns the reason for failure.
+// The fresh token is persisted if either the access token or refresh token has changed.
 func (uotm *UserOAuthTokenManager) getCachedTokenInfo(ctx context.Context) (*OAuthTokenInfo, error) {
-	hasToken, err := uotm.credCache.HasCachedToken()
+
+	// retrieveRecord gets an AuthenticationRecord from the local disk
+	record, err := retrieveRecord()
 	if err != nil {
-		return nil, fmt.Errorf("no cached token found, please log in with azcopy's login command, %v", err)
```

Comment:
If retrieve record fails then we still need to try out credCache as that might be the legit source of token here.

Reply:
Retrieve record's implementation is such that if  Authentication-record does not exist it will return an empty Authentication-record otherwise it fetches the existing record and tries to create the Authentication-record by un-marshaling[parsing the json record to create Authentication-record] this error is returned in the case where un-marshaling was at failure otherwise we are looking in the credcache basis of the empty record found.